packages/core/src/dom_components/model/ComponentDynamicValueListener.ts

```diff
@@ -0,0 +1,123 @@
+import { ObjectAny } from '../../common';
+import EditorModel from '../../editor/model/Editor';
+import Component, { dynamicAttrKey } from './Component';
+import { DynamicValueWatcher } from './DynamicValueWatcher';
+
+export class ComponentDynamicValueListener {
+  private propertyWatcher: DynamicValueWatcher;
+  private attributeWatcher: DynamicValueWatcher;
+  private traitsWatcher: DynamicValueWatcher;
+
+  constructor(
+    private component: Component,
+    em: EditorModel,
+  ) {
+    this.propertyWatcher = new DynamicValueWatcher(this.createPropertyUpdater(), em);
+    this.attributeWatcher = new DynamicValueWatcher(this.createAttributeUpdater(), em);
+    this.traitsWatcher = new DynamicValueWatcher(this.createTraitUpdater(), em);
+  }
+
+  private createPropertyUpdater() {
+    return (key: string, value: any) => {
+      this.component.set(key, value, { skipWatcherUpdates: false });
```

Comment:
I think when we update the component from data sources changes, we don't want this change to be counted as a dirty change and skip it from the undoManager (`avoidStore` option should already handle both cases).

Probably DataSource itself could define if it needs to trigger the dirty change and if the change is undoable.
For example, a data source with external records (eg. REST API) could retrigger the fetch of all records on a specific interval. The editor would update components with new data but I'd not expect changes on the project data itself.
In case of static data source records (records stored in the project JSON), their update will trigger the dirty change and could be defined as an undoable action.